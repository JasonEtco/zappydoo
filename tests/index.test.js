const fs = require('fs')
const path = require('path')
const Zappydoo = require('../lib')

describe('zappydoo', () => {
  let data, templates

  beforeEach(() => {
    templates = {
      basic: fs.readFileSync(path.join(__dirname, 'fixtures', 'template.md'), 'utf8'),
      missing: fs.readFileSync(path.join(__dirname, 'fixtures', 'template-missing.md'), 'utf8')
    }

    data = { park: 'Fenway', weather: 'sunny', user: 'Jason' }
  })

  describe('precompile', () => {
    let zap

    beforeEach(() => {
      zap = new Zappydoo()
    })

    it('should return a function', () => {
      const renderer = zap.precompile(templates.basic)
      expect(renderer.toString()).toMatchSnapshot()
    })

    it('should be able to run the function later', () => {
      const renderer = zap.precompile(templates.basic)
      expect(renderer(data)).toMatchSnapshot()
    })
  })

  describe('render', () => {
    let zap

    beforeEach(() => {
      zap = new Zappydoo()
    })

    it('should render the correct string', () => {
      const arr = ['Hello and welcome to ', ' park! The weather is ', ' today. Enjoy your day ', '!']
      const map = new Map()
      map.set(1, 'park')
      map.set(3, 'weather')
      map.set(5, 'user')

      const rendered = zap.render(arr, map, data)
      expect(rendered).toMatchSnapshot()
    })

    it('throws if the property does not exist', () => {
      const map = new Map()
      map.set(1, 'park')
      map.set(3, 'adjective')
      map.set(5, 'weather')
      map.set(8, 'user')

      try {
        zap.render([], map, data)
      } catch (e) {
        expect(e.message).toMatchSnapshot()
      }
    })
  })

  describe('compile', () => {
    it('should return the correct string', () => {
      const compiled = Zappydoo.compile(templates.basic, data)
      expect(compiled).toMatchSnapshot()
    })

    it('leaves undefined variables blank', () => {
      const compiled = Zappydoo.compile(templates.missing, data)
      expect(compiled).toMatchSnapshot()
    })
  })
})
