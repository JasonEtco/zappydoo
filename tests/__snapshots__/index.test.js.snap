// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zappydoo compile leaves undefined variables blank 1`] = `"Hello and welcome to Fenway park! The weather is  sunny today. Enjoy your day Jason!"`;

exports[`zappydoo compile should return the correct string 1`] = `"Hello and welcome to Fenway park! The weather is sunny today. Enjoy your day Jason!"`;

exports[`zappydoo precompile should be able to run the function later 1`] = `"Hello and welcome to Fenway park! The weather is sunny today. Enjoy your day Jason!"`;

exports[`zappydoo render should render the correct string 1`] = `"Hello and welcome to Fenway park! The weather is sunny today. Enjoy your day Jason!"`;

exports[`zappydoo render throws if the property does not exist 1`] = `"The property adjective does not exist in the provided object."`;
